### PR TITLE
[SHELL32] Follow-up of #2076

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-16533](https://jira.reactos.org/browse/CORE-16533)

BEFORE:
before

AFTER:
after

### CODE REVIEW

**Comment by milawynsrealm** [comment]
Should this also apply to *.com programs, or is the icon handled another way?

**Comment by katahiromz** [comment]
`*.com` has already the default icon. It doesn't have a file-specific icon.

**Comment by learn-more** [comment]
@katahiromz According to a comment from @yagoulas on mattermost this is a hack, and not the correct place to implement this functionality:

> Katayama Hirofumi MZ : in a file that has 0 logic for handling icons depending on file type you added a change that changes the icon based on the name of file. Doesn't that ring any bell that it is not the right place to put this check?
>
>like, in the place where all the logic to decide the icon of a file already exists?
>
>because right now, it is split into two places. and not only that, CExtractIcon is completely content agnostic, it can handle all sorts of icons, not only file icons. stuff like control panel, my computer, everything. and in this generic thing we now have a check for a specific kind of shell items, for a specific kind of files

From https://chat.reactos.org/reactos/pl/33eghdicxpr1jywib8dpsa96zw

**Comment by katahiromz** [comment]
@yagoulas Your opinion is wrong.

This is a test program:
https://jira.reactos.org/secure/attachment/55354/ExtractTest.zip

The results:
ExtractTest

`IExtractIcon::GetIconLocation` is not content agnostic.

**Comment by katahiromz** [comment]
after
Still working...

File: dll/win32/shell32/folders/CFSFolder.cpp [hunk 1]
```diff
@@ -244,6 +244,31 @@ getIconLocationForFolder(IShellFolder * psf, PCITEMID_CHILD pidl, UINT uFlags,
...
+    HMODULE hModule,
+    LPCWSTR lpszType,
+    LPWSTR lpszName,
+    LONG_PTR lParam)
+{
+    *(BOOL *)lParam = TRUE;
+    return FALSE;
+}
+
+static BOOL ExeFileHasAnyIcon(LPCWSTR pszFile)
```
Comment (learn-more) [inline dll/win32/shell32/folders/CFSFolder.cpp]: You can use `ExtractIconExW` for this, with a `-1` nIconIndex:
> Specifies the zero-based index of the first icon to extract. For example, if this value is zero, the function extracts the first icon in the specified file.
>
> **If this value is –1 and phiconLarge and phiconSmall are both NULL, the function returns the total number of icons in the specified file**. If the file is an executable file or DLL, the return value is the number of RT_GROUP_ICON resources. If the file is an .ico file, the return value is 1.

Comment (katahiromz) [reply dll/win32/shell32/folders/CFSFolder.cpp]: I found our `ExtractIconExW` returns `1` even if exe has no icon. It seems we cannot use `ExtractIconExW` for this purpose.

**Pull request merged**